app/src/main/java/pl/agh/eye/CameraActivity.kt

```diff
@@ -0,0 +1,143 @@
+package pl.agh.eye
+
+import android.Manifest
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.util.Log
+import android.view.MenuItem
+import android.view.SurfaceView
+import android.view.View
+import android.view.WindowManager
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import org.opencv.android.*
+import org.opencv.android.CameraBridgeViewBase.CvCameraViewFrame
+import org.opencv.android.CameraBridgeViewBase.CvCameraViewListener2
+import org.opencv.core.Core
+import org.opencv.core.CvType
+import org.opencv.core.Mat
+import org.opencv.imgproc.Imgproc
+
+
+class CameraActivity : AppCompatActivity(), CvCameraViewListener2 {
+    // Loads camera view of OpenCV for us to use. This lets us see using OpenCV
+    private var mOpenCvCameraView: CameraBridgeViewBase? = null
+    // Used in Camera selection from menu (when implemented)
+    private val mIsJavaCamera = true
+    private val mItemSwitchCamera: MenuItem? = null
+    // These variables are used (at the moment) to fix camera orientation from 270degree to 0degree
+    var mRgba: Mat? = null
+    var mRgbaF: Mat? = null
+    var mRgbaT: Mat? = null
+    private val mLoaderCallback: BaseLoaderCallback = object : BaseLoaderCallback(this) {
+        override fun onManagerConnected(status: Int) {
+            when (status) {
+                LoaderCallbackInterface.SUCCESS -> {
+                    Log.i(
+                        TAG,
+                        "OpenCV loaded successfully"
+                    )
+                    mOpenCvCameraView!!.enableView()
+                }
+                else -> {
+                    super.onManagerConnected(status)
+                }
+            }
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        Log.i(TAG, "called onCreate")
+        super.onCreate(savedInstanceState)
+        checkAndRequestPermissions()
+        window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
+        setContentView(R.layout.show_camera)
+        mOpenCvCameraView =
+            findViewById<View>(R.id.show_camera_activity_java_surface_view) as JavaCameraView
+        mOpenCvCameraView!!.visibility = SurfaceView.VISIBLE
+        mOpenCvCameraView!!.setCvCameraViewListener(this)
+        mOpenCvCameraView!!.enableView()
+    }
+
+    public override fun onPause() {
+        super.onPause()
+        if (mOpenCvCameraView != null) mOpenCvCameraView!!.disableView()
```

Comment:
takiego czegoś też w kotlinie raczej nie. Takie proste nullchecki super załatwia operator `?.`
Czyli to będzie `mOpenCvCameraView?.disableView()`